src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/CreateMergedManifestBuildModelTests.cs

```diff
@@ -0,0 +1,289 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using FluentAssertions;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.DotNet.Maestro.Tasks.Proxies;
+using Microsoft.DotNet.VersionTools.BuildManifest.Model;
+using Moq;
+using NUnit.Framework;
+
+namespace Microsoft.DotNet.Maestro.Tasks.Tests
+{
+    [TestFixture]
+    class CreateMergedManifestBuildModelTests
+    {
+        private static readonly string initialAssetsLocation = "thisIsALocation";
+        private static readonly int azDoBuildId = 12345;
+        private static readonly int azDoBuildDefId = 67890;
+        private static readonly string azDoAccount = "thisIstheAzDoAccount";
+        private static readonly string azDoProject = "thisIsTheAzDoProject";
+        private static readonly string azDoBuildNumber = "thisIsAnAzDoBuildNumberFromTheManifest";
+        private static readonly string azDoRepo = "thisIsARepoFromTheManifest";
+        private static readonly string azDoBranch = "thisIsAnAzDoBranch";
+        private static readonly int publishingVersion = 1234567890;
+        private static readonly string isReleasePackage = "false";
+
+        private static readonly string buildRepoName = "thisIsARepo";
+        private static readonly string buildNumber = "azDevBuildNumber";
+        private static readonly string sourceBranch = "thisIsASourceBranch";
+        private static readonly string commitSourceVersion = "thisIsASourceVersion";
+
+        private PushMetadataToBuildAssetRegistry pushMetadata;
+        private BuildModel expectedBuildModel;
```

Comment:
This shouldn't be a field. Every test is messing around with it's properties.

Tests should call some "Get me all the fake stuff I need" method that builds them and returns them, rather than depending on [Setup] to set a bunch of fields.

If not, these need non-parallelisable on them. But I'd rather we not write state tests with side effects, they can be extremely difficult to track failures down in, because the test might only pass/fail because of stateful changes some other test made.

Reply:
Done